core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -223,6 +223,8 @@ object NonEmptyVector extends NonEmptyVectorInstances {
     new NonEmptyVector(buf.result)
   }
 
+  def unapply[A](nev: NonEmptyVector[A]): Some[(A, Vector[A])] = Some((nev.head, nev.tail))
```

Comment:
@larsrh Got a code snippet showing that?

AFAIK Some can't check exhaustiveness any more than Option as it's not sealed (sealedness isn't transitive nor does final doesn't imply sealed, double sadly).